### DISCUSSION TITLE
xmake自动生成的compile_commands.json编译器为gcc而不是clang

### DISCUSSION THREAD

**Discussion by yeangPan** [xmake-io/xmake#5705] [General]
我的cpp环境为VSCode + Clangd + XMake + LLDB。

settings.json如下：
```
add_rules("mode.debug", "mode.release")
set_languages("c99","c++20")
set_defaultmode("debug")
set_targetdir("$(projectdir)/build")
add_requires("spdlog")


-- 指定使用 Clang 编译器
-- set_toolset("cc", "clang")
-- set_toolset("cxx", "clang++")

target("mytinywebserver")
    set_kind("binary")
    add_files("src/*.cpp")
    add_packages("spdlog")
```
我想请教一下，为什么在不指定clang编译器的情况下，生成的```compile_commands.json ```里面编译器是gcc而不是clang。xmake configuration中的  ```compile commands backend ```是```clangd ```：

```
[
{
  "directory": "/home/xxx/projects/cpp/mytinywebserver",
  "arguments": ["/usr/bin/gcc", "-c", "-m64", "-fvisibility=hidden", "-fvisibility-inlines-hidden", "-O3", "-std=c++20", "-isystem", "/home/yangp/.xmake/packages/s/spdlog/v1.14.1/469f312ffeff47d9aa249e3ed8e400dc/include", "-DNDEBUG", "-o", "build/.objs/mytinywebserver/linux/x86_64/release/src/main.cpp.o", "src/main.cpp"],
  "file": "src/main.cpp"
}]

```

**Comment by star-hengxing** [comment]
`compile_commands` 和编译命令是一样的。
不需要单独切换工具链，用 `xmake f -c --toolchain=clang` 或 `set_toolchains("clang")`

**Reply by yeangPan** [reply]
vscode的插件配置xmake configuration，compile commands backend 是clangd 这里设置的是什么配置项。另外上面的问题，那个是xmake.lua  配置 ，不是settings.json。settings.json如下：
```
{
    "clangd.arguments": [
        "--compile-commands-dir=.vscode",
    ],
}
```

**Comment by waruqi** [comment]
> 我想请教一下，为什么在不指定clang编译器的情况下，生成的compile_commands.json 里面编译器是gcc而不是clang。

没有为什么，gcc 和 clang 同时存在的情况下，目前默认优先会去找 gcc ，你又没配置指名非要 clang

如果你想要 clang ，那就按上面说的，自己切到 clang toolchain 去。

> xmake configuration中的 compile commands backend 是clangd ：

那个只是 xmake-vscode 那个插件自动设置了 clangd 作为 lsp 的 backend 而已，跟编译器用哪个 没有必然关系
